deepfence_frontend/apps/dashboard/src/features/settings/pages/UserManagement.tsx

```diff
@@ -1,13 +1,271 @@
+import { Suspense, useCallback, useMemo, useState } from 'react';
+import { IconContext } from 'react-icons';
+import { FaPencilAlt, FaTrashAlt } from 'react-icons/fa';
+import { HiOutlineExclamationCircle } from 'react-icons/hi';
+import {
+  ActionFunctionArgs,
+  generatePath,
+  Link,
+  useFetcher,
+  useLoaderData,
+} from 'react-router-dom';
+import { toast } from 'sonner';
+import { Button, createColumnHelper, Modal, Table, TableSkeleton } from 'ui-components';
+
+import { getUserApiClient } from '@/api/api';
+import { ApiDocsBadRequestResponse } from '@/api/generated';
+import { ModelUser } from '@/api/generated/models/ModelUser';
 import { SettingsTab } from '@/features/settings/components/SettingsTab';
+import { ApiError, makeRequest } from '@/utils/api';
+import { typedDefer, TypedDeferredData } from '@/utils/router';
+import { DFAwait } from '@/utils/suspense';
+import { usePageNavigation } from '@/utils/usePageNavigation';
+
+type LoaderDataType = {
+  message?: string;
+  data?: ModelUser[];
+};
+const getUsers = async (): Promise<LoaderDataType> => {
+  const usersPromise = await makeRequest({
+    apiFunction: getUserApiClient().getUsers,
+    apiArgs: [],
+  });
+
+  if (ApiError.isApiError(usersPromise)) {
+    return {
+      message: 'Error in getting users list',
+    };
+  }
+
+  return {
+    data: usersPromise,
+  };
+};
+const loader = async (): Promise<TypedDeferredData<LoaderDataType>> => {
+  return typedDefer({
```

Comment:
there are typescript errors with the typeDefer usages, lets meet and sort this out.